src/minimetrics/transport.py

```diff
@@ -0,0 +1,110 @@
+import re
+from typing import Dict, Generic, List, Optional, TypeVar
+
+import sentry_sdk
+from sentry_sdk.envelope import Envelope, Item
+
+from minimetrics.types import ExtractedMetric, ExtractedMetricValue, MetricTagsInternal, MetricUnit
+from sentry.utils import metrics
+
+IN = TypeVar("IN")
+OUT = TypeVar("OUT")
+M = TypeVar("M")
+
+sanitization_re = re.compile(r"[^a-zA-Z0-9_/.]+")
+
+
+class EncodingError(Exception):
+    pass
+
+
+class MetricEnvelopeEncoder(Generic[IN, OUT]):
+    def encode(self, value: IN) -> OUT:
+        raise NotImplementedError()
+
+
+class RelayStatsdEncoder(MetricEnvelopeEncoder[ExtractedMetric, str]):
+    MULTI_VALUE_SEPARATOR = ":"
+    TAG_SEPARATOR = ","
+
+    def encode(self, value: ExtractedMetric) -> str:
+        metric_name = self._sanitize_value(value["name"])
+        if not metric_name:
+            raise EncodingError("The sanitized metric name is empty")
+        metric_unit = self._get_metric_unit(value.get("unit"))
+        metric_value = self._get_metric_value(value["value"])
+        metric_type = value["type"]
+        metric_tags = self._get_metric_tags(value.get("tags"))
+        metric_timestamp = self._get_metric_timestamp(value["timestamp"])
+
+        return f"{metric_name}{metric_unit}:{metric_value}|{metric_type}{metric_tags}{metric_timestamp}"
+
+    def _sanitize_value(self, value: str) -> str:
+        # Remove all non-alphanumerical chars which are different from _ / .
+        return sanitization_re.sub("", value)
```

Comment:
_nit_: This will remove dashes (`-`) entirely, where we should probably replace them with underscores, or simply allow dashes. Let's keep this for a follow-up to define the legal character set and implement it across all services in one go.

Reply:
Filed under https://github.com/orgs/getsentry/projects/149/views/7?pane=issue&itemId=38348338